File: game-of-life-cpp/Guarana.cpp
#include "include/Guarana.h"

Guarana::Guarana(World* world, Point& position)
    : Plant(world, 0, position, "🍒", 0) {
}

Guarana::Guarana(World* world, Point& position, int age)
    : Plant(world, 0, position, "🍒", age) {
}

Guarana::~Guarana() {

}

int Guarana::collision(Organism& other) {
    if(this->getStrength() < other.getStrength()) {
        other.setStrength(other.getStrength() + 3);
        return 0;
    }
    return 0;
}

File: game-of-life-java/src/Guarana.java
import javax.swing.*;
import java.awt.*;

public class Guarana extends Plant {
    private static final ImageIcon guaranaIcon = new ImageIcon("images/guarana.png");
    public static final Image scaledGuaranaIcon = guaranaIcon.getImage().getScaledInstance(30, 30, Image.SCALE_SMOOTH);

    public Guarana(Point position, World world, int age) {
        super(0, position, world, age);
    }

    @Override
    public ImageIcon getIcon() {
        return guaranaIcon;
    }

    @Override
    public void collision(Organism other) {
        other.setStrength(other.getStrength() + 3);
       String message = "Organism " + other.getClass().getSimpleName() + " consumed Guarana and gained +3 strength!";
       world.getWindow().addMessage(message);
    }
}
